spring-data-relational/src/test/java/org/springframework/data/relational/degraph/DependencyTests.java

```diff
@@ -37,6 +38,7 @@ public void cycleFree() {
 				classpath() //
 						.noJars() //
 						.including("org.springframework.data.relational.**") //
+						.excluding("org.springframework.data.relational.core.dialect.**") //
```

Comment:
Please do not change this test.
The change also should be superfluous once the `.../core/sql` no longer references `Dialect`.

Reply:
Yeah, I agree with you, but, when I have applied changes in this PR and supply them with proper tests, this particular test method cause the build to fail. I will try it out again and if I will encounter the same behavior again, I will reach you out.